front/src/modules/ui/components/board/Board.tsx

```diff
@@ -33,85 +22,39 @@ export interface Column {
   itemKeys: BoardItemKey[];
 }
 
-type BoardProps = {
-  initialBoard: Column[];
-  items: Items;
-};
-
-export const Board = ({ initialBoard, items }: BoardProps) => {
-  const [board, setBoard] = useState<Column[]>(initialBoard);
-
-  const onDragEnd: OnDragEndResponder = useCallback(
-    (result) => {
-      const { destination, source } = result;
-      if (!destination) return;
-      const sourceColumnIndex = board.findIndex(
-        (column) => column.id === source.droppableId,
-      );
-      const sourceColumn = board[sourceColumnIndex];
-      const destinationColumnIndex = board.findIndex(
-        (column) => column.id === destination.droppableId,
-      );
-      const destinationColumn = board[destinationColumnIndex];
-      if (!destinationColumn || !sourceColumn) return;
-      const sourceItems = sourceColumn.itemKeys;
-      const destinationItems = destinationColumn.itemKeys;
-
-      const [removed] = sourceItems.splice(source.index, 1);
-      destinationItems.splice(destination.index, 0, removed);
-
-      const newSourceColumn = {
-        ...sourceColumn,
-        itemKeys: sourceItems,
-      };
-
-      const newDestinationColumn = {
-        ...destinationColumn,
-        itemKeys: destinationItems,
-      };
-
-      const newBoard = [...board];
-      newBoard.splice(sourceColumnIndex, 1, newSourceColumn);
-      newBoard.splice(destinationColumnIndex, 1, newDestinationColumn);
-      setBoard(newBoard);
-    },
-    [board],
+export const getOptimisticlyUpdatedBoard = (
```

Comment:
I wonder if we should even extract this to another file modules/dnd, this would remove the dependency from ui/board to DND completely

Reply:
We'd still have dependencies on the types "DropableProvided"